File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.java
class Solution {
    int diameter=0;
    public int height(TreeNode root)
    {
     if(root==null) return 0;
      int lh=height(root.left);
      int rh=height(root.right);
        diameter=Math.max(diameter,lh+rh);
      return Math.max(lh,rh)+1;
    }
    public int diameterOfBinaryTree(TreeNode root) {
        //int diameter=0;
        height(root);
        return diameter;
    }
}

File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.cpp
class Solution {
public:
    int diameterOfBinaryTree(TreeNode* root) {
        int diameter=0;
        height(root,diameter);
        return diameter;
    }
    int height(TreeNode *root, int& diameter)
    {
        if(!root)
            return 0;
        int lh=height(root->left,diameter);
        int rh=height(root->right,diameter);
        diameter=max(diameter,lh+rh);
        return max(lh,rh)+1;
    }
};